environs/__init__.py

```diff
@@ -276,29 +276,74 @@ def read_env(
         file is found. If you do not wish to recurse up the tree, you may pass
         False as a second positional argument.
         """
-        # By default, start search from the same file this function is called
-        if path is None:
+        # declare constants
+        PATH_SEPARATOR_PATTERN = r"\\|\/"
+
+        # ensure that path is not a directory
+        if path and os.path.isdir(str(path)):
+            raise ValueError(
+                "The specified path '" + str(path) + "' is a directory. Ensure that you pass a filename."
+            )
+
+        # remember original path
+        original_path = path
+
+        # use .env as path if path is None or empty
+        if not path:
+            path = ".env"
+
+        # make path absolute if it is relative, using the directory of the calling script as base
+        if not os.path.isabs(path):
+
+            def merge_absolute_and_relative_path(absolute_path, relative_path):
+                absolute_path_items = re.split(PATH_SEPARATOR_PATTERN, absolute_path)
+                relative_path_items = re.split(PATH_SEPARATOR_PATTERN, relative_path)
+                popped = False
+                for absolute_path_item in absolute_path_items:
+                    if absolute_path_item == relative_path_items[0]:
+                        relative_path_items.pop(0)
+                        popped = True
+                    elif popped:
+                        break
+                return os.path.sep.join(absolute_path_items + relative_path_items)
+
             current_frame = inspect.currentframe()
             if not current_frame:
                 raise RuntimeError("Could not get current call frame.")
             frame = typing.cast(types.FrameType, current_frame.f_back)
             caller_dir = os.path.dirname(frame.f_code.co_filename)
-            # Will be a directory
-            start = os.path.join(os.path.abspath(caller_dir))
+            path = merge_absolute_and_relative_path(caller_dir, path)
+
+        # walk up the directory tree starting in the path directory and try to find an environment file.
+        # if recurse is False, stop after the first directory.
+        def look_for_environment_file_in_parents(path):
```

Comment:
Is this necessary? Can this be handled by `load_dotenv` (or some other function in `dotenv`)?

Reply:
I think so. I am not aware of an existing function which does the same.